components/style/animation.rs

```diff
@@ -507,7 +507,8 @@ where Damage: TRestyleDamage {
                                                                  frame);
             if updated_style {
                 if let Some(damage) = damage {
-                    *damage = *damage | Damage::compute(Some(style), &new_style);
+                    *damage =
+                        *damage | Damage::compute_for_layout(Some(style), &new_style);
```

Comment:
Let's return the new style and let the caller handle it (or, if that doesn't work, let the caller provide a callback that gets passed the new style), and then remove the distinction between compute_for_layout and compute_for_cascade.